### ISSUE TITLE
Storybook: Remove "experimental" category in favor of a badge

### ISSUE THREAD

**Issue by tyxla** [WordPress/gutenberg#66805]
In #66275 we're currently in the process of reorganizing the components by categories. 

However, we separate components by stable and experimental already, and as a result, we get multiple directories with the same category:

Stable:
![Image](https://github.com/user-attachments/assets/0e248b78-b548-4c6b-8d40-9c0c3d4fb115)

Experimental:
![Image](https://github.com/user-attachments/assets/9af25ec6-9e47-4b4b-bd72-9466a33ebadf)

Ideally, we shouldn't have that multi-level distinction by category, and the categories would be at the top level. 

To achieve that, I think we can mark the experimental components with a badge, just like we currently mark WIP or private components.

cc @WordPress/gutenberg-components

**Comment by ciampo** [comment]
Related conversation in https://github.com/WordPress/gutenberg/issues/66197

**Comment by yogeshbhutkar** [comment]
I noticed that this issue still has the `Discussion` label. Before moving forward with a PR, I’d like to confirm whether it’s still relevant and requires a PR.

cc: @mirka @ciampo

**Comment by ciampo** [comment]
We should be probably good to go, IMO, but let's also hear confirmation from @mirka to make sure

**Comment by mirka** [comment]
Yes, I'm good to move forward. How about we use the 🧪 emoji for the badge.
